lib/puppet/pops/parser/egrammar.ra

```diff
@@ -361,7 +361,7 @@ if_expression
       }
     | ELSE LBRACE statements RBRACE {
         result = Factory.block_or_expression(val[2])
-        loc result, val[0], val[3]
+        loc result, val[2][0], val[3]
```

Comment:
This is problematic since the block_or_expression may be empty, and val[2][0] would then be nil and that would fail this call. Also, it is not right that the location for the else expression is the first expression in the  body, the location should cover the entire length from the first letter in 'else' to the closing brace '}'.